implants/lib/pb/src/config.rs

```diff
@@ -106,18 +108,17 @@ impl Config {
         let beacon_id =
             std::env::var("IMIX_BEACON_ID").unwrap_or_else(|_| String::from(Uuid::new_v4()));
 
-        #[cfg(feature = "dns")]
-        let transport = crate::c2::beacon::Transport::Dns;
-        #[cfg(feature = "http1")]
-        let transport = crate::c2::beacon::Transport::Http1;
-        #[cfg(feature = "grpc")]
-        let transport = crate::c2::beacon::Transport::Grpc;
-        #[cfg(not(any(feature = "dns", feature = "http1", feature = "grpc")))]
-        let transport = crate::c2::beacon::Transport::Unspecified;
+        let transport_type = match CALLBACK_URI.split(":").nth(0).unwrap_or("unspecified") {
```

Comment:
Probably need to add some testing to make sure the transport exists in the complied agent.